mlir/utils/performance/quickTunerPreproc.py

```diff
@@ -0,0 +1,169 @@
+
+
+Note:
+If using MITuna edit MITuna/tuna/rocmlir/rocmlir_worker.py, editing:
+
+
+    cmd = env_str + f" python3 ./bin/tuningRunner.py -q {special_args} \
+                     --config='{config_string}' --mlir-build-dir `pwd` \
+                     --output=- --tflops \
+                     --rocmlir_gen_flags='--device={self.gpu_id}' 2>/dev/null"
+
+
+to:
+    
+    import uuid
+
+    if not os.path.exists("./run"):
+      os.makedirs("./run")
+
+    unique_file_id = uuid.uuid4().hex
+
+    file_id = os.path.join("./run", unique_file_id)
+    
+    cmd = env_str + f" python3 ./bin/tuningRunner.py -q {special_args} \
+                     --config='{config_string}' --mlir-build-dir `pwd` \
+                     --output={file_id} --tflops --debug \
+                     --rocmlir_gen_flags='--device={self.gpu_id}' 2>/dev/null"
+
+"""
+
+import os
+import sys
+import argparse
+import pandas as pd
+import glob
+from sklearn.preprocessing import MinMaxScaler
+
+class qtPreprocessor(object):
+    """
+    class to process *.debug files into a single script
+    """
+
+    def __init__(self, pargs):
+        self.input_dir = pargs.input_dir
+
+    @staticmethod
+    def __get_stats_gemm(df, ct):
+        """
+        static helper method to get stats for a dataframe:
+        (number of files processed, number of unique gemms, group by datatype)
+        """
+        print(f"Files processed: {ct}")
+
+        # num of dtypes
+        dtypes= {t[0]:df for t,df in df.groupby(['DataType'])}
+
+        print("Types found:")
+        for dt in dtypes:
+            print(f"\t{dt}")
+
+        # num unique gemms in file:
+        cols = ['TransA', 'TransB', 'G', 'M', 'N','K']
+        unique_gemms = df[cols].drop_duplicates()
+        
+        num_gemms = len(unique_gemms)
+        print(f"Number of unique Gemms: {num_gemms}")
+        for _,row in unique_gemms.iterrows():
+            tup = tuple(row)
+            print(f"{tup[0]},{tup[1]},{tup[2]},{tup[3]},{tup[4]},{tup[5]}")
+        
+
+    @staticmethod
+    def __get_stats_conv(df, ct):
+        """
+        static helper method to get stats for a dataframe:
+        (number of files processed, number of unique gemms, group by datatype)
+        """
+        raise NotImplementedError()
+        
+    @staticmethod
+    def process(input_dir, output_name=None, op='gemm', file_ext="debug", debug=False, normalize=True):
+        """
+        staticmethod process() function that compiles output files into a single dataframe and saves to tsv file
+        """
+
+        tsv_files = glob.glob(os.path.join(input_dir, f"*.{file_ext}"))
+        print(os.path.join(input_dir, f"*.{file_ext}"))
+
+        dfs = []
+        ct = 0
+        for file in tsv_files:
+            df = pd.read_csv(file, sep='\t')
+            if normalize:
+                scaler = MinMaxScaler()
+                df['TFlops'] = scaler.fit_transform(df[['TFlops']])
```

Comment:
Why one set of brackets on the left but two on the right??

Reply:
I think that was a mess up on my end. Fortunately (or unfortunately) the syntax is forgiving. 

Also, this will actually be changed to add a `NormalizedTFlops` column instead of overwriting the current one.
```
if normalize:
    scaler = MinMaxScaler()
    df['NormalizedTFlops'] = scaler.fit_transform(df[['TFlops']])
```